mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/CenterIntegrationTest.java

```diff
@@ -81,5 +88,51 @@ public void testFullCenterCreation() {
         Assert.assertTrue(center.getExternalId().equals(externalId));
         Assert.assertTrue(center.getStaffId() == staffId);
         Assert.assertTrue(center.isActive() == false);
+        int[] groupMemberList = new int[center.getGroupMembers().size()];
+        for (int i = 0; i < groupMemberList.length; i++) {
+            groupMemberList[i] = ((Double) center.getGroupMembers().get(i).get("id")).intValue();
+        }
+
+        Assert.assertArrayEquals(groupMemberList, groupMembers);
+    }
+
+    @Test
+    public void testListCenters() {
+        ArrayList<CenterDomain> paginatedList = CenterHelper.paginatedListCenters(requestSpec, responseSpec);
+        ArrayList<CenterDomain> list = CenterHelper.listCenters(requestSpec, responseSpec);
+
+        Assert.assertNotNull(paginatedList);
+        Assert.assertNotNull(list);
+        Assert.assertTrue(Arrays.equals(paginatedList.toArray(new CenterDomain[paginatedList.size()]),
+                list.toArray(new CenterDomain[list.size()])));
+    }
+
+    @Test
+    public void testCenterRetrieval() {
+        int id = CenterHelper.listCenters(requestSpec, responseSpec).get(0).getId();
+        Assert.assertTrue(id > 0);
+
+        CenterDomain center = CenterHelper.retrieveByID(id, requestSpec, responseSpec);
+        Assert.assertNotNull(center.getName());
+        Assert.assertNotNull(center.getHierarchy());
+        Assert.assertNotNull(center.getOfficeName());
+    }
+
+    @Ignore
+    @Test
+    public void testVoidCenterRetrieval() {
+        ArrayList<CenterDomain> arr = CenterHelper.listCenters(requestSpec, responseSpec);
+        long id = arr.get(arr.size() - 1).getId();
+        BatchRequest br = new BatchRequest();
+
+        br.setRequestId(id);
+        br.setRelativeUrl("centers");
+        br.setMethod("GET");
+        br.setBody("{ }");
+
+        List<BatchResponse> response = BatchHelper.postWithSingleRequest(requestSpec, responseSpec, br);
```

Comment:
Sorry for being unclear. Instead of using Batch stuff simply create a ResponseSpec that expects a 404 instead of 200.